src/main/java/com/sergiofah/model/Category.java

```diff
@@ -25,4 +27,16 @@ public class Category {
 
     @OneToMany(mappedBy = "category")
     private List<Product> products;
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) return true;
+        if (object == null) return false;
+
+        if(object.getClass() == CategoryDTO.class){
+            CategoryDTO categoryDTO = (CategoryDTO) object;
+            return this.id.equals(categoryDTO.getId()) && (this.category.equals(categoryDTO.getCategory())) && (this.line.getId().equals(categoryDTO.getLineId()));
+        }
```

Comment:
encontrei dois pequenos pontos para correção e melhoria também:
1. Favor, não usar o operador "==" quando for comparar dois objetos, apenas quando for números. Isso porque o operador "==" mesmo sendo mais rápido (porém não é notório) ele não tem uma assertividade tão boa quanto usando o método equals(). Isso pode até ser explicado para o caso de sobrescrever o método `equals()`, imagine caso o método o assertArrayEquals usasse o operador "==" ao invés do método equals dos objetos que ele recebe como parâmetro.
2. Quanto ao retorno, sugiro realizar uma quebra de linha para melhorar a visualização do código e deve ser feita da maneira correta, para isso, sugiro dar uma lida sobre as convenções do Java, ele é importante pois cada linguagem possui uma maneira correta na hora de quebrar linhas, como escrever uma variavel e assim por diante. (https://www.oracle.com/java/technologies/javase/codeconventions-introduction.html)